ci/run_conditional_tests.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# `-e` enables the script to automatically fail when a command fails
+# `-o pipefail` sets the exit code to the rightmost comment to exit
+# with a non-zero
+set -eo pipefail
+
+export PROJECT_ROOT=$(realpath $(dirname "${BASH_SOURCE[0]}")/..)
+
+# A script file for running the test in a sub project.
+test_script="${PROJECT_ROOT}/ci/run_single_test.sh"
```

Comment:
What is the intent of "single" in `run_single_test`? It looks like it's meant to run a single _type_ of test. Might be nice to say that here.

Also, this script relies on the `BUILD_TYPE` env var, and `run_single_test` relies on `TEST_TYPE`? It might be nice to list all the environment variables used by each script and its dependencies.

Reply:
Fixed in 1bf8bab